### PR TITLE
Gamecore interaction functions with relevant unit test

### LINKED ISSUE #1: init dev env
1. import https://github.com/SailGame/proto   (use submodule)
2. import gtest/gmock,   grpc (submodule or fetchcontent or whatever it is)
3. set up git action (optional)
4. set up clang-format
5. set up CMakeList skeleton.

### PR DESCRIPTION
**Pull request by MrHate**
Anyone got a *script*?

### CODE REVIEW

File: src/Dummy.cpp [hunk 1]
```diff
@@ -10,6 +10,8 @@ const Dummy::uid_t Dummy::Join(const std::string &addr) {
   ++user_count;
   uid2addr.emplace(LastPlayer(), addr);
   holecards.emplace(LastPlayer(), std::vector<card_t>());
+  bankroll.emplace(LastPlayer(), 0);
```
Comment (Veiasai) [inline src/Dummy.cpp]: curious,  couldn't you make the player as a complicate class?
e.g. 
class player {
    mUid
    mbankRoll
    roundBets
}

**Review by Veiasai — COMMENTED** [review]
go go go.
Import CommonLib now.

**Comment by MrHate** [comment]
> go go go.
> Import CommonLib now.

An enough complete gamecore is necessary before that :)

**Comment by Gusabary** [comment]
> go go go.
> Import CommonLib now.

wait a moment, pls import new CommonLib directly, which has some refactor about `StateMachine` and is coming soon

**Review by Veiasai — APPROVED** [review]

**Comment by MrHate** [comment]
I still have 3 work to do:
1. ~I shall test how Dummy behaves with just 2 players alive next;~
2. ~All-in liquidation；~
3. ~And what if there are multiple winners within a single game?~

**Comment by MrHate** [comment]
Need to test liquidation solely I think.

**Pull request merged**